src/stare/auth.py

```
@@ -0,0 +1,434 @@
+        self._exchanged_expires_at: int = 0
+        # Locking: thread lock guards in-process races; file lock guards
+        # cross-process races (e.g. two CLI invocations refreshing at once).
+        self._thread_lock = threading.Lock()
+        self._file_lock = filelock.FileLock(self._storage.lock_path)
+
+    @property
+    def token_path(self) -> Path:
+        """Path to the stored token JSON file."""
+        return self._token_path
+
+    def login(
```

Ruff (0.15.10)

[warning] Too many statements (52 > 50)

(PLR0915)